File: Day05/Part01/day05_01.java
import java.io.BufferedReader;
import java.io.FileReader;
import java.util.ArrayList;


class day05_01 {


    public static void main (String[] args) {

        // get array of all integer values
        Integer[] jumps = getJumps("input.txt");

        // while in range, keep traversing the array
        int jump = 0;
        int position = 0;
        int loopCount = 0;
        while (position >= 0 && position < jumps.length) {
            loopCount++;
            jump = jumps[position];
            jumps[position] += 1;
            position += jump;
        }

        // print answer
        System.out.println("Number of Loops: " + loopCount);

    }


    public static Integer[] getJumps(String filename) {
        
        try {
            BufferedReader br = new BufferedReader(new FileReader(filename));
            ArrayList<Integer> int_list = new ArrayList<Integer>();
            String currentIntString = br.readLine();
            Integer currentInt = null;
            while (currentIntString != null) {
                currentInt = Integer.parseInt(currentIntString);
                int_list.add(currentInt);
                currentIntString = br.readLine();
            }
            return int_list.toArray(new Integer[int_list.size()]);
        } catch (Exception e) {
            e.printStackTrace();
            return null;
        }

    }


}

File: Day05/Part01/day05_01.py
def get_jumps():
    input_file = open('input.txt', 'r')
    jumps = []
    next_number = input_file.readline()
    while (next_number):
        jumps.append(int(next_number))
        next_number = input_file.readline()
    input_file.close
    return jumps


if __name__ == '__main__':

    # get list of all jump sizes
    jumps = get_jumps()

    next_jump = 0
    jump_count = 0
    current_pos = 0

    # stop looping after jumping out of range
    while (0 <= current_pos < len(jumps)):     
        next_jump = jumps[current_pos]
        jump_count += 1
        jumps[current_pos] += 1
        current_pos += next_jump

    # print answer
    print("Number of Loops:", jump_count)
